sql-plugin/src/main/scala/com/nvidia/spark/rapids/ColumnarPartitionReaderWithPartitionValues.scala

```diff
@@ -95,6 +110,55 @@ object ColumnarPartitionReaderWithPartitionValues {
     result
   }
 
+  /**
+   * Splits each Gpu Column Vector into multiple batches based on row counts.
+   *
+   * Example,
+   * columnVectors = [ Cv('pqr', 55), Cv('111', 55), Cv('ab1', 55)]
+   * rowCounts = [ 20, 10, 15, 10 ]
+   * splitIndices = [ 20, 30, 35 ]
+   * Result:
+   *    [
+   *      [Cv('pqr',20), Cv('pqr',10), Cv('pqr',15), Cv('pqr',10)],
+   *      [Cv('111',20), Cv('111',10), Cv('111',15), Cv('111',10)],
+   *      [Cv('ab1',20), Cv('ab1',10), Cv('ab1',15), Cv('ab1',10)]
```

Comment:
The documentation for this method is a bit lacking and the example doesn't help clarify.  The one-liner doc says it splits based on row counts, and that's what the example shows, but then what is splitIndices doing?  The example looks like one could just ignore it and get the right answer.

Reply:
> splitIndices are indices calculated from rowCounts using cumulative sum

That's not true in the example, hence why it's so confusing.  20+10 is indeed 30, but 30+15 is not 35.